src/main/java/de/adesso/anki/battle/GameEngine.java

```diff
@@ -145,34 +184,83 @@ private void updateSimulation(long deltaNanos) {
            body.updatePosition(deltaNanos);
         }
     }
+	
+	
+    private void collectOrphanedWeapons() {
+    	Iterator<Body> it = world.getBodiesModifiable().iterator();
+        while(it.hasNext()){
+        	Body weapon = it.next();
+        	if (weapon instanceof Rocket && ((Rocket) weapon).shouldExplode()) {
+        		it.remove();
+        		System.out.println("Delete orphaned rocket");

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)